magento/pwa-studio
# [PWA-245] Shipping Information (Authenticated)

## Description

As an authenticated user, during checkout my saved profile information should be auto-populated during the checkout process. Information including: 

- Edit contact info: should prepopulate with saved info 
- Edit shipping address (display of addresses in address book and entering a new address and applying the change)
- default address book should be selected for shipping address
- If no address exists within address book (or default is selected) then standard shipping form fields should display
Important:

Do not display temp data from shipping estimate fields. Compare gql response to `MOCKED_ADDRESS` to verify.

#### Notes
Notes from a team discussion on the use cases associated with this issue - for your reference Tommy Wiebell

- If the shopper does not have a default address saved, the "add default address" button would take them directly to the address form in a dialog and right back to the checkout review page once they were done adding the address

- All addresses added will be saved to the address book by default and the shopper will be able to remove them from the address book in the account section if they so choose
- In the case that saved addresses exist, and the shopper goes from the edit link > list of addresses and adds a new one, they will go directly back to the checkout review screen after adding a new address and bypass the list of addresses, since we think it is safe to assume that they meant to use the address they just added to checkout with
- The list of addresses would be presented in a "tab" and the 'add new address' form in a "dialog"
On mobile, it needs to be apparent that the 'add new address' form is an overlay on the addresses list "tab"

## Related Issue





* [[PWA-245](https://jira.corp.magento.com/browse/PWA-245)] Shipping Information (Authenticated)

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


Going to outline all the different flows that will need to be tested. Can flesh out more details if it isn't clear what needs to be entered.
1. Guest Entry / Edit (verify no regressions)
2. Authenticated User - No Default Address: Form displays inline similar to Guest. New message block about entry being automatically added to book and saved as default. There should be no checkbox to not proceed that way.
3. Authenticated User - With Default Address: Default is auto selected and proceed to shipping method step.
4. Authenticated Edit Flow: Now takes you too full screen page for Address Book

Address Book Flows
1. Edit Address: Similar to Guest edit, new set as default checkbox, no email field, no messages. Edit addresses, flag one as new default, verify information is persisted on save.
2. Add Address: Similar to above but new address. Verify both add as default, and add in general.
3. Select Address: Select a new address and hit Apply. Verify view goes back to checkout with selected address applied.

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
